src/gobtest/data_consistency/data_consistency_test.py

```diff
@@ -59,29 +79,47 @@ def run(self):
         success = 0
         missing = 0
 
-        for row in rows:
+        gob_count = self._get_gob_count()
 
-            if cnt % test_every == random_offset:
-                gob_row = self._get_matching_gob_row(row)
+        with ProgressTicker(f"Compare data ({gob_count:,})", 10000) as progress:
+            for row in rows:
+                progress.tick()
 
-                if not gob_row:
-                    seqnr = f' and volgnummer {row.get(FIELD.SEQNR)}' if self.has_states else ''
-                    logger.warning(f"Row with id {self._get_row_id(row)}{seqnr} missing")
-                    missing += 1
-                else:
-                    success += int(self._validate_row(row, gob_row))
-                checked += 1
+                # Always test the first row, test other rows at random
+                if cnt == 0 or cnt % test_every == random_offset:
+                    gob_row = self._get_matching_gob_row(row)
+
+                    if not gob_row:
+                        seqnr = f' and volgnummer {row.get(FIELD.SEQNR)}' if self.has_states else ''
+                        logger.warning(f"Row with id {self._get_row_id(row)}{seqnr} missing")
+                        missing += 1
+                    else:
+                        success += int(self._validate_row(row, gob_row))
+                    checked += 1
+
+                cnt += 1
+
+        self._log_result(checked, cnt, gob_count, missing, success)
 
-            cnt += 1
+    def _log_result(self, checked, cnt, gob_count, missing, success):
+        if gob_count != cnt:
+            logger.error(f"Counts don't match: source {cnt:,} - GOB {gob_count:,} ({abs(cnt - gob_count):,})")
```

Comment:
Maybe include a threshold here as well? We're currently using a threshold for missing rows, but with this check that threshold won't do much.

When n rows below the threshold are missing no error is logged there, but in that case the count should not match and an error is generated anyway. I think it should be one or the other; either check for the exact count (and remove the missing threshold), or accept some missing rows, but also accept those missing rows in this check.

I'm not sure what the correct way to go is; I've added the missing threshold to account for updates between the IMPORT and the DUMP, but if that rarely happens we could remove the missing threshold too. If it happens often I think it's best to use a threshold for the count as well.

Reply:
Well, I do not get any errors on the count when I run it for the full BRK. So I propose to leave the exact comparison for now. Probably for the other catalogs it need to be adjusted.

Reply:
Discussed on Slack; removal of missing_threshold added to BAG ticket.